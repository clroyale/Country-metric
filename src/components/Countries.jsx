import React from 'react';
import Country from './Country';
import styles from '../styles/Countries.module.css';
import imagePath from '../assets/world_map.png';

function Countries() {
  const Data = [
    {
      id: '1',
      name: 'Baba yara',
      capacity: 4500,
    },
    {
      id: '2',
      name: 'Accra Sports Stadium',
      capacity: 4500,
    },
    {
      id: '3',
      stadium: 'Camp Nou',
      capacity: 100000,
    },
    {
      id: '4',
      name: 'Signa Iduna Park',
      capacity: 9800,
    },
  ];
  return (
    <div className={styles.main_container}>
      {Data.map((country) => (
        <Country
          key={country.id}
          country={country.name}
          capacity={country.capacity}
          image={imagePath}
        />
      ))}
    </div>
  );
}

export default Countries;
