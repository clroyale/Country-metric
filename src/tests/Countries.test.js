import { useSelector } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Countries from '../components/Countries';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Countries', () => {
  const mockData = [
    {
      name: { common: 'country 1' }, population: 1234, flags: { png: 'image-link.png' }, coatOfArms: { png: 'image-link.png' }, id: '1',
    },
    {
      name: { common: 'country 2' }, population: 1200, flags: { png: 'image-link.png' }, coatOfArms: { png: 'image-link.png' }, id: '2',
    },
    {
      name: { common: 'country 3' }, population: 1534, flags: { png: 'image-link.png' }, coatOfArms: { png: 'image-link.png' }, id: '3',
    },
  ];

  beforeEach(() => {
    useSelector.mockReturnValue({ countries: mockData });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correctly', () => {
    const tree = renderer.create(
      <Router>
        <Countries />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should display all countries on load', () => {
    render(
      <Router>
        <Countries />
      </Router>,
    );
    const countryDisplayed = screen.getAllByRole('link', { name: /^country/i });
    expect(countryDisplayed).toHaveLength(3);
  });

  it('Should display searched country', () => {
    render(
      <Router>
        <Countries />
      </Router>,
    );
    const searchInput = screen.getByPlaceholderText('enter country name to search');
    fireEvent.change(searchInput, { target: { value: 'country 3' } });

    const countryDisplayed = screen.getAllByRole('link', { name: /^country/i });
    expect(countryDisplayed).toHaveLength(1);
  });
});
