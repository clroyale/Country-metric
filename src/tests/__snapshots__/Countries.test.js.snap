// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries Should render correctly 1`] = `
<div
  className="main_container"
>
  <div
    className="search_input_container"
  >
    <input
      className="search_input"
      id=""
      name=""
      onChange={[Function]}
      placeholder="enter country name to search"
      type="search"
      value=""
    />
  </div>
  <div
    className="countries_container"
  >
    <a
      className="main_container"
      href="/details/country 1"
      onClick={[Function]}
    >
      <img
        alt="country 1-flag"
        src={null}
      />
      <div
        className="country_mini_info"
      >
        <h2
          className="country_name"
        >
          country 1
        </h2>
        <p
          className="capacity"
        >
          1234
        </p>
      </div>
    </a>
    <a
      className="main_container"
      href="/details/country 2"
      onClick={[Function]}
    >
      <img
        alt="country 2-flag"
        src={null}
      />
      <div
        className="country_mini_info"
      >
        <h2
          className="country_name"
        >
          country 2
        </h2>
        <p
          className="capacity"
        >
          1200
        </p>
      </div>
    </a>
    <a
      className="main_container"
      href="/details/country 3"
      onClick={[Function]}
    >
      <img
        alt="country 3-flag"
        src={null}
      />
      <div
        className="country_mini_info"
      >
        <h2
          className="country_name"
        >
          country 3
        </h2>
        <p
          className="capacity"
        >
          1534
        </p>
      </div>
    </a>
  </div>
</div>
`;
